dds/idl/dds_generator.h

```diff
@@ -517,7 +518,7 @@ void generateBranchLabels(AST_UnionBranch* branch, AST_Type* discriminator,
       has_default = true;
     } else if (discriminator->node_type() == AST_Decl::NT_enum) {
       if (use_id) {
-        const unsigned id = be_global->get_id(NULL, branch, 0);
+        const unsigned id = be_global->get_id(NULL, branch, default_id + j);
```

Comment:
This will also need the union. Also I don't understand what the last argument is doing. As far as I know the labels have nothing to do with the member id of the branch.